q2_feature_table/tests/filter/test_filter_sequences.py

```diff
@@ -0,0 +1,102 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import unittest
+
+import qiime2
+import pandas as pd
+from pandas.util.testing import assert_series_equal
+
+from q2_feature_table import filter_seqs
+
+
+class FilterSeqsTests(unittest.TestCase):
+    def setUp(self):
+        self.seqs = pd.Series(['ACGT', 'GCTA', 'CCCC', 'TGTT'],
+                              index=['O1', 'O2', 'O3', 'O4'])
+        self.md_lite = pd.DataFrame(
+            [['A'], ['C'], ['G'], ['T']], index=['O1', 'O2', 'O3', 'O4'],
+            columns=['seq'])
+        md_full = pd.DataFrame(
+            [['foo', '1'], ['bar', '2'], ['baz', '3'], ['foo', '4']],
+            index=['O1', 'O2', 'O3', 'O4'], columns=['stuff', 'some_numbers'])
+        md_full.index.name = 'FeatureID'
+        self.md_full = qiime2.Metadata(md_full)
```

Comment:
It would be good to see a metadata object where none of the IDs matched at all.

Reply:
There are a couple of examples of that in the tests ([with](https://github.com/qiime2/q2-feature-table/pull/140/files#diff-179288ed1a08aed39629f8e221d51606R82) and [without](https://github.com/qiime2/q2-feature-table/pull/140/files#diff-179288ed1a08aed39629f8e221d51606R48) `where`) - maybe I am misunderstanding the request, though.

Reply:
I was thinking more along the lines of the metadata has nothing in common with the artifact (vs an empty filter from where). I get that they'll hit the same code path, but it seems like the most common way to mess this up.

Reply:
Isn't that the test linked above under [`without`](https://github.com/qiime2/q2-feature-table/pull/140/files#diff-179288ed1a08aed39629f8e221d51606R48)? That metadata has nothing in common with the artifact.

Reply:
I apologize, I only looked at the `with`. My bad!